modules/oadp-api-details.adoc

```diff
@@ -0,0 +1,164 @@
+// Module included in the following assemblies:
+//
+// * oadp-api
+
+:_content-type: REFERENCE
+[id="oadp-api-details_{context}"]
+= OADP API
+
+The following tables expose the structure of the OADP API:
+
+.DataProtectionApplicationSpec
+[options="header"]
+|===
+|Property|Type|Description
+
+|`BackupLocations`
+|[] `BackupLocation`
+|Defines the list of desired configurations to use for `BackupStorageLocations`.
+
+|`SnapshotLocations`
+|[] `SnapshotLocation`
+|Defines the list of desired configurations to use for `VolumeSnapshotLocations`.
+
+|`UnsupportedOverrides`
+|map [ `UnsupportedImageKey` ] string
+|Can be used to override the deployed dependent images for development.
+
+|`PodAnnotations`
+|map [ string ] string
+|Used to add annotations to pods deployed by Operators.
+
+|`PodDnsPolicy`
+|[`corev1.DNSPolicy`] link:https://pkg.go.dev/k8s.io/api/core/v1#DNSPolicy[DNSPolicy]
+|Defines how the configuration of the DNS of a pod.
+
+|`PodDnsConfig`
+|[`corev1.PodDNSConfig`] link:https://pkg.go.dev/k8s.io/api/core/v1#PodDNSConfig[PodDNSConfig]
+|Defines the DNS parameters of a pod in addition to those generated from `DNSPolicy`.
+
+|`BackupImages`
+|*bool
+|Used to specify whether or not you want to deploy a registry for enabling backup and restore of images.
+
+|`Configuration`
+|*`ApplicationConfig`
+|Used to define the data protection application's server configuration.
+|===
```

Comment:
Probably missed data mover because our reference docs in https://github.com/openshift/oadp-operator/blob/master/docs/API_ref.md still isn't updated. @shubham-pampattiwar @eemcmullan @savitharaghunathan please update upstream as well. Once this PR is reviewed we could consider redirecting everything to downstream.

Reply:
Added.